contracts/BC_fusion/AirDrop.sol

```diff
@@ -0,0 +1,245 @@
+
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./System.sol";
+import "./lib/Utils.sol";
+
+/// @title AirDrop is used to claim the token from BC users.
+/// @dev This is designed for the BC users to claim the token from TokenHub.
+/// The BC will chain will stop and generate a merkle tree root after BC-fusion plan was started.
+/// The BC users can claim the token from TokenHub after the merkle tree root is generated.
+/// For more details, please refer to the BEP-299(https://github.com/bnb-chain/BEPs/pull/299).
+contract AirDrop is IAirDrop, ReentrancyGuardUpgradeable, System {
+    using Utils for string;
+    using Utils for bytes;
+
+    /*----------------- init paramters -----------------*/
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approverAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bool public merkleRootAlreadyInit = false;
+
+    /*----------------- storage -----------------*/
+    /// claimedMap is used to record the claimed token.
+    mapping(bytes32 => bool) private claimedMap;
+
+    /*----------------- permission control -----------------*/
+    /// assetProtector is the address that is allowed to pause the claim.
+    address public assetProtector = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    /// blackList is the address that is not allowed to claim.
+    mapping(address => bool) public blackList;
+    /// paused is used to pause the claim.
+    bool private _paused;
+
+    modifier whenNotPaused() {
+        if (_paused) revert AirDropPaused();
+        _;
+    }
+
+    modifier onlyAssetProtector() {
+        if (msg.sender != assetProtector) revert OnlyAssetProtector();
+        _;
+    }
+
+    modifier notInBlackList() {
+        if (blackList[msg.sender]) revert InBlackList();
+        _;
+    }
+
+    modifier merkelRootReady() {
+        if (!merkleRootAlreadyInit) revert MerkleRootNotInitialize();
+        if (merkleRoot == bytes32(0)) revert MerkleRootNotInitialize();
+        _;
+    }
+
+    function pause() external onlyAssetProtector {
+        _paused = true;
+        emit Paused();
+    }
+
+    function resume() external onlyAssetProtector {
+        _paused = false;
+        emit Resumed();
+    }
+
+    function addToBlackList(address account) external onlyAssetProtector {
+        blackList[account] = true;
+    }
+
+    function removeFromBlackList(address account) external onlyAssetProtector {
+        blackList[account] = false;
+    }
+
+    /*----------------- events -----------------*/
+    /// This event is triggered whenever a call to #pause succeeds.
+    event Paused();
+    /// This event is triggered whenever a call to #pause succeeds.
+    event Resumed();
+    /// This event is triggered whenever a call to #claim succeeds.
+    event Claimed(bytes32 tokenSymbol, address account, uint256 amount);
```

Comment:
Is it still good to name as `Claimed`, because the fund is pending but not claimed.

Reply:
change to => `TokenRecoverRequestd`